### PR TITLE
fix: OneHotEncoder no longer creates duplicate column names

### LINKED ISSUE #201: `OneHotEncoder` can accidentally create columns with same name
### Describe the bug

The `OneHotEncoder` uses the schema `<old_column_name>_<value>` to name the created columns. This can lead to conflicts, however.

### To Reproduce

Run this program:

```py
from safeds.data.tabular.containers import Table
from safeds.data.tabular.transformation import OneHotEncoder

if __name__ == '__main__':
    table = Table.from_dict({"a_b": ["c"], "a": ["b_c"]})
    transformed_table = OneHotEncoder().fit_and_transform(table)

    print(transformed_table)
```

It raises an exception:

```
ValueError: Length mismatch: Expected axis has 2 elements, new values have 1 elements
```

The issue is that two columns with the same name (`a_b_c`) get created.

### Expected behavior

No exception. The names of all created columns should be unique. They should also not conflict with existing columns in the `Table`. This can be done by detecting conflicts between two created columns or between a created column and an existing, unchanged column and appending a suffix `_<counter>` to the names of the created columns (e.g. `a_b_c_1` vs. `a_b_c_2`).

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by zzril**
Closes #201.

### Summary of Changes

Changed OneHotEncoder to manually implement the encoding.
(Breaking) Changed the format of newly generated columns to use two underscores as separator. In case of naming conflicts, a hash and a unique ID will be appended to the column name.

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.2) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/4891865470)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.2/descriptors/python_black)          |    1|    1|                                                             0|0.6s        |
|❌ PYTHON    |[mypy](https://megalinter.io/6.22.2/descriptors/python_mypy)            |    1|     |[7](https://github.com/Safe-DS/Stdlib/actions/runs/4891865470)|1.87s       |
|❌ PYTHON    |[ruff](https://megalinter.io/6.22.2/descriptors/python_ruff)            |    1|    1|[4](https://github.com/Safe-DS/Stdlib/actions/runs/4891865470)|0.04s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.2/descriptors/repository_git_diff)|yes  |     |no                                                            |0.03s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4891865470)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/tabular/transformation/_one_hot_encoder.py [hunk 1]
```diff
@@ -119,37 +134,49 @@ def transform(self, table: Table) -> Table:
...
         # Input table does not contain all columns used to fit the transformer
         missing_columns = set(self._column_names.keys()) - set(table.column_names)
         if len(missing_columns) > 0:
             raise UnknownColumnNameError(list(missing_columns))
 
-        original = table._data.copy()
-        original.columns = table.schema.column_names
+        # Make a copy of the table:
+        # TODO: change to copy method once implemented
+        new_table = table.remove_columns([])
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_one_hot_encoder.py:147]: Is the copy really necessary? At first glance, none of the operations you call on `new_table` mutate its internal state.

File: src/safeds/data/tabular/transformation/_one_hot_encoder.py [hunk 2]
```diff
@@ -174,21 +201,29 @@ def inverse_transform(self, transformed_table: Table) -> Table:
...
         # Transformer has not been fitted yet
-        if self._wrapped_transformer is None or self._column_names is None:
+        if self._column_names is None or self._value_to_column is None:
             raise TransformerNotFittedError
 
-        data = transformed_table._data.copy()
-        data.columns = transformed_table.column_names
+        # Make a copy of the table:
+        # TODO: change to copy method once implemented
+        new_table = transformed_table.remove_columns([])
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_one_hot_encoder.py:209]: Is the copy really necessary? At first glance, none of the operations you call on `new_table` mutate its internal state.

**Comment by lars-reimann** [comment]
Once #301 is implemented, we can shorten the implementation of the `OneHotEncoder` a little.

File: src/safeds/data/tabular/transformation/_one_hot_encoder.py (hunk 1 above)
Comment (zzril) [reply src/safeds/data/tabular/transformation/_one_hot_encoder.py]: You're right, the copy is not neccessary. Removed it.

We do, however, need to make sure the column count can't temporarily drop to 0 (that would confuse the `table` class), so I moved the part where we remove the old columns to the bottom.

**Pull request merged**